### ISSUE TITLE
if() statement before calling random generators for note probability and random pitch

### ISSUE THREAD

**Issue by oddtime** [hydrogen-music/hydrogen#1088]
in `src/core/Hydrogen.cpp` line 980 circa

```
float rnd = (float)rand()/(float)RAND_MAX;
if (pNote->get_probability() < rnd) {
m_songNoteQueue.pop();
pNote->get_instrument()->dequeue();
continue;
}
```

```
// Offset + Random Pitch ;)
pNote->set_pitch( pNote->get_pitch()
			+ pNote->get_instrument()->get_pitch_offset()
			+ getGaussian( 0.4 ) * pNote->get_instrument()->get_random_pitch_factor() );
```

the random generators could be called after a check if noteprobability and randomPitch are different from 1 and 0.

**Comment by oddtime** [comment]
see pull req #1104

**Comment by theGreatWhiteShark** [comment]
Closed with #1104
